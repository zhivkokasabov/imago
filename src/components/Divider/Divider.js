import React from 'react';

import classNames from 'classnames';

import styles from './Divider.module.scss';

export default function Button({ className, variant = 'dark' }) {
  return (
    <div className={classNames(styles[`divider__${variant}`], styles.divider, className)} />
  );
}