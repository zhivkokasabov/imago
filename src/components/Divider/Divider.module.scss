.divider {
  width: 100%;
  height: 1px;

  &__light {
    background-color: rgb(var(--coolgrey-50));
  }

  &__dark {
    background-color: #000;
  }
}