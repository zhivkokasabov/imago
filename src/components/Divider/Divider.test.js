import React from 'react';
import { render } from '@testing-library/react';

import Divider from './Divider';

describe('Divider component', () => {
  it('should render Divider', () => {
    const { asFragment } = render(<Divider />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render light Divider', () => {
    const { asFragment } = render(<Divider variant="light" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
