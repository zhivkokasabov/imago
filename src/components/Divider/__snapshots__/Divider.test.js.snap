// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component should render Divider 1`] = `
<DocumentFragment>
  <div
    class="divider__dark divider"
  />
</DocumentFragment>
`;

exports[`Divider component should render light Divider 1`] = `
<DocumentFragment>
  <div
    class="divider__light divider"
  />
</DocumentFragment>
`;
